tests/lib/__init__.py

```diff
@@ -492,10 +492,7 @@ def __init__(self, base_path, *args, **kwargs):
         kwargs.setdefault("cwd", self.scratch_path)
 
         # Setup our environment
-        environ = kwargs.get("environ")
-        if environ is None:
-            environ = os.environ.copy()
-
+        environ = kwargs.setdefault("environ", os.environ.copy())
```

Comment:
This no longer has an effect?

Reply:
It actually does, because is it equivalent to
```python
environ = kwargs.get('environ', os.environ.copy())
kwargs['environ'] = environ
```
and `kwargs` is to be passed to `TestFileEnvironment` (L510 of the new revision).  Now that you're asking, I'm not sure if this line is really obvious (as in, easy to read and interpret).  Also the current test suit doesn't seem to pass `environ` to `PipTestEnvironment` at all&mdash;do you want me to switch to solely `environ = os.environ.copy()` and pass that to `super().__init__` explicitly?

Reply:
Ah OK, you’re modifying `kwargs['environ']` in-place. This is not very straightforward IMO (in-place modification to a nested item generally isn’t), but I guess I’m fine here since code here does not need to be read often.